gpcc-mocks/src/main/java/uk/nhs/adaptors/gpccmocks/gpc/GpcController.java

```diff
@@ -0,0 +1,208 @@
+package uk.nhs.adaptors.gpccmocks.gpc;
+
+import static org.springframework.http.HttpStatus.ACCEPTED;
+
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.getHostAndPortFromRequest;
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.getResponseHeaders;
+import static uk.nhs.adaptors.gpccmocks.common.ControllerHelpers.isUuid;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.badRequest;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.invalidNhsNumber;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.patientNotFound;
+import static uk.nhs.adaptors.gpccmocks.common.OperationOutcomes.referenceNotFound;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.jayway.jsonpath.JsonPath;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gpccmocks.common.TemplateUtils;
+
+@RestController
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GpcController {
+    @PostMapping(value = "/{odsCode}/STU3/1/gpconnect/structured/fhir/Patient/$gpc.getstructuredrecord")
+    @ResponseStatus(value = ACCEPTED)
+    public ResponseEntity<String> accessStructuredRecord(
+        HttpServletRequest request,
+        @PathVariable String odsCode,
+        @RequestHeader(name = "Ssp-TraceID") String sspTraceId,
+        @RequestHeader(name = "Ssp-From") String sspFrom,
+        @RequestHeader(name = "Ssp-To") String sspTo,
+        @RequestHeader(name = "Ssp-InteractionID") String sspInteractionId,
+        @RequestHeader(name = HttpHeaders.AUTHORIZATION) String authorization,
+        @RequestBody String requestBody) {
+
+        var host = getHostAndPortFromRequest(request);
+
+        log.debug("Request for 'Access Structured Record'. " +
+                "odsCode={} Ssp-TraceID={} Ssp-From={} Ssp-To={} Ssp-InteractionID={} Host/X-Forwarded-Host: {}",
+            odsCode, sspTraceId, sspFrom, sspTo, sspInteractionId, host);
+
+        log.debug("Request body for 'Access Structured Record'\n{}", requestBody);
+
+        if (!isUuid(sspTraceId)) {
+            return badRequest("Ssp-TraceID header must be a UUID");
```

Comment:
gpc is responding with Operation Outcome 

```
{
    "resourceType": "OperationOutcome",
    "meta": {
        "profile": [
            "https://fhir.nhs.uk/STU3/StructureDefinition/GPConnect-OperationOutcome-1"
        ]
    },
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "details": {
                "coding": [
                    {
                        "system": "https://fhir.nhs.uk/STU3/CodeSystem/Spine-ErrorOrWarningCode-1",
                        "code": "BAD_REQUEST",
                        "display": "BAD_REQUEST"
                    }
                ]
            },
            "diagnostics": "Ssp-InteractionID header blank"
        }
    ]
}
```

instead of:

```
{
    "timestamp": "2021-06-28T09:37:31.373+00:00",
    "status": 400,
    "error": "Bad Request",
    "message": "",
    "path": "/B82617/STU3/1/gpconnect/structured/fhir/Patient/$gpc.getstructuredrecord"
}
```

that happens to all missing headers. They are required by default.

Reply:
The mock doesn't capture the spec completely. I don't think this difference will block and further development work. Could you create a backlog ticket for the improvement?

Reply:
ok

Reply:
https://gpitbjss.atlassian.net/browse/NIAD-1566